url-matchers.go

```diff
@@ -122,7 +123,22 @@ func unique[T comparable](items []T) []T {
 // and a function to actually do the matching and producing of the *URL
 type URLMatcher struct {
 	Type string
-	Fn   func(*Node) *URL
+	Fn   func(*Node) []*URL
+}
+
+func SingleURLMatcher(typ string, fn func(*Node) *URL) URLMatcher {
+	wrapperFn := func(n *Node) []*URL {
+		result := fn(n)
+		if result == nil {
+			return nil
+		}
+		return []*URL{result}
+	}
+	return URLMatcher{Type: typ, Fn: wrapperFn}
```

Comment:
I have considered three ways to implement multiple urls:
- `URLMatcher` interface with `URLMatcherSingle` and `URLMatcherMulti` implementation (seemed too complicated to me for the current feature, but it is possible not to break backward compatibility)
- New `FnMulti` field in `URLMatcher` (seemed dirty to me because it is unclear how to handle cases when one or more functions are passed; this one also breaks backward compatibility since it's impossible to init struct without explicit field names - `URLMatcher{"string", func(){...}} -> URLMatcher{Type: "string", Fn: func(){...}}`)
- this appoach with single field, but updated function signature

From these three approaches, the last one seemed to me the most convenient.